src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php

```diff
@@ -99,7 +105,20 @@ public function getArguments(Request $request, callable $controller, ?\Reflectio
                 }
             }
 
-            throw new \RuntimeException(sprintf('Controller "%s" requires that you provide a value for the "$%s" argument. Either the argument is nullable and no null value has been provided, no default value has been provided or there is a non-optional argument after this one.', $this->getPrettyName($controller), $metadata->getName()));
+            $reasons = array_map(static fn (NearMissValueResolverException $e) => $e->getMessage(), $valueResolverExceptions);
+            if (empty($reasons)) {
```

Comment:
```suggestion
            if (!$reasons) {
```

Reply:
Any particular reason for the suggestion other than personal preference? :)

Reply:
I had seen that `empty()` is essentially the concise equivalent to `!isset($var) || $var == false`.  `empty()` really necessary in this case? https://www.php.net/manual/en/function.empty.php